cdap-common/src/main/java/io/cdap/cdap/internal/app/store/RunRecordDetail.java

```diff
@@ -187,34 +187,48 @@ public String toString() {
   /**
    * @return Builder to create a RunRecordDetail
    */
-  public static Builder builder() {
-    return new Builder();
+  public static ABuilder builder() {
+    return new ABuilder();
   }
 
   /**
    * @param record existing record to copy fields from
    * @return Builder to create a RunRecordDetail, initialized with values from the specified existing record
    */
-  public static Builder builder(RunRecordDetail record) {
-    return new Builder(record);
+  public static ABuilder builder(RunRecordDetail record) {
+    return new ABuilder(record);
   }
 
   /**
    * Builds RunRecordMetas.
    */
-  public static class Builder extends RunRecord.Builder<Builder> {
-    private ProgramRunId programRunId;
-    private String twillRunId;
-    private Map<String, String> systemArgs;
-    private byte[] sourceId;
-    private String principal;
-    private ArtifactId artifactId;
-
-    private Builder() {
+  public static class ABuilder extends Builder<ABuilder> {
```

Comment:
Could you please switch the names? A prefix usually indicates an abstract class to inherit from, so you should have `Builder extends ABuilder`, not visa versa. And say in `ABuilder` that it's present for children builders to inherit from

Reply:
ack